tests/test_meminfo_hierarchy.sh.in

```diff
@@ -19,22 +19,45 @@ cleanup() {
 FAILED=1
 trap cleanup EXIT HUP INT TERM
 
-[ ! -d /sys/fs/cgroup/memory ] && exit 0
-echo "==> Setting up memory cgroup"
-initmemory=`awk -F: '/memory/ { print $3 }' /proc/1/cgroup`
-mempath=/sys/fs/cgroup/memory/${initmemory}
+IS_CGROUP_V2=0
+cat /proc/1/mountinfo | grep "cgroup cgroup" > /dev/null || IS_CGROUP_V2=1
+
+if [ "$IS_CGROUP_V2" = "1" ]; then
+    [ ! -d /sys/fs/cgroup ] && exit 0
+    echo "==> Setting up cgroup"
+    mempath=/sys/fs/cgroup/
+
+    memory_limit_file=memory.max
+    tasks_file=cgroup.procs
+else
+    [ ! -d /sys/fs/cgroup/memory ] && exit 0
+    echo "==> Setting up memory cgroup"
+    initmemory=`awk -F: '/memory/ { print $3 }' /proc/1/cgroup`
```

Comment:
I've converted to `$(.)` syntax.

About matching I'm no t sure, just because in cgroup-v1 a few controllers can be in one line of `/proc/*/cgroup`.

Thanks, Simon!